mealie/services/backups_v2/alchemy_exporter.py

```diff
@@ -39,23 +41,33 @@ def __init__(self, connection_str: str) -> None:
         self.session_maker = sessionmaker(bind=self.engine)
 
     @staticmethod
-    def convert_to_datetime(data: dict) -> dict:
+    def is_uuid(value: str) -> bool:
+        try:
+            uuid.UUID(value)
+            return True
+        except ValueError:
+            return False
+
+    def convert_types(self, data: dict) -> dict:
         """
-        walks the dictionary to convert all things that look like timestamps to datetime objects
+        walks the dictionary to restore all things that look like string representations of their complex types
         used in the context of reading a json file into a database via SQLAlchemy.
         """
         for key, value in data.items():
             if isinstance(value, dict):
-                data = AlchemyExporter.convert_to_datetime(value)
+                data = self.convert_types(value)
             elif isinstance(value, list):  # assume that this is a list of dictionaries
-                data[key] = [AlchemyExporter.convert_to_datetime(item) for item in value]
+                data[key] = [self.convert_types(item) for item in value]
             elif isinstance(value, str):
-                if key in AlchemyExporter.look_for_datetime:
-                    data[key] = AlchemyExporter.DateTimeParser(dt=value).dt
-                if key in AlchemyExporter.look_for_date:
-                    data[key] = AlchemyExporter.DateTimeParser(date=value).date
-                if key in AlchemyExporter.look_for_time:
-                    data[key] = AlchemyExporter.DateTimeParser(time=value).time
+                if self.is_uuid(value):
+                    # convert the data to the current database's native GUID type
+                    data[key] = GUID.convert_value_to_guid(value, self.engine.dialect)
```

Comment:
This is the line that fixes this nightmare

Reply:
???? How did you figure this out?

Reply:
Pain

Reply:
The actual answer though: I stared at a lot of compiled queries and eventually noticed that all of my GUIDs were lowercase and had no dashes. Then I peeked into my backup data and noticed all of my GUIDs had dashes. Then I threw a hail-mary and bypassed our GUID implementation (see: https://github.com/mealie-recipes/mealie/pull/2859/commits/5cf5abe27966bbc0d2e9e8a13976c2e5b5c234a7) and my restores started working

Looking closer at the GUID implementation, we keep the dashes in postgres (i.e. we keep the literal GUID representation as a string) and we convert the GUIDs to 32 char hex in sqlite. So sqlite GUIDs, in our implementation, are assumed to have no dashes. But our backup restore doesn't strip them out if they originally came from postgres